tests/test_alloc.rs

```diff
@@ -57,10 +57,10 @@ mod tests {
 
         assert_eq!(
             count_alloc(|| {
-                let _ = TestDeku::try_from(input.as_ref()).unwrap();
+                let _ = TestDeku::from_reader((&mut input.as_slice(), 0)).unwrap();
             })
             .0,
-            (4, 0, 4)
+            (5, 0, 5)
```

Comment:
This should come from [this](https://github.com/sharksforarms/deku/pull/352/files#diff-267c9f4da66412a9f439ac08d224356fe24265b5e1cebb6c44c2d55b96414513R177). We basically need to store the bits that weren't used into `reader.leftover`. This currently allocates. 

__Although__ we should be able to just keep around a BitArray of [bool; 8] or something like that and store them just in the struct since I don't think it would ever keep around more then 8. It's a good TODO for later, especially as it most likely will remove most perf regressions for reading only bits.